### ISSUE TITLE
Prototype a project manager role with our current team

### ISSUE THREAD

**Issue by choldgraf** [2i2c-org/team-compass#398]
### Context

In recent conversations (see #382 for more context) we've agreed that having somebody serve in a coordinator / manager role would help our team be more efficient and effective at getting our work done. 
Getting a new person hired in this role will take some time, and in the meantime we would benefit from having some improved processes as a team.

So, I think we should prototype a "service manager" role with our current team. This will give us an opportunity to try out some of the role aspects described [in the role draft document](https://docs.google.com/document/d/1voDLLolN-2szln_KzdpNBH6aaobE8ajcjWYfv4c0DNk/edit?usp=sharing), and should help us make some quick improvements in team coordination.

In addition, I spoke with @damianavila and he would be willing to reduce his engineering capacity to 50% in order to dedicate the rest of his time to serving in this role.

### Proposal

I propose that @damianavila serves as 2i2c's first "service manager", at a 50% capacity. We'll use [the role description](https://docs.google.com/document/d/1voDLLolN-2szln_KzdpNBH6aaobE8ajcjWYfv4c0DNk/edit?usp=sharing) as a guide for what he'll focus his time on. In the immediate future, this means things like:

- Facilitating our team planning meetings (e.g. bi-weekly sprint planning)
- Overseeing and improving our team planning and coordination processes
- Overseeing and improving our "Site Reliability Engineering" practices
- Working with team members to unblock and facilitate their work

In the short term, I don't think we need to necessarily change our team processes - but this role could help us follow them a bit more closely, and put more thought into iterating and improving upon them in the short term.

### Questions to answer during the trial

One of the goals of this trial run is to answer questions that will help us make decisions about how to staff this role moving forward. Here are a few major ones to keep in mind, that we can add to as we learn more:

1. How much work does this role entail? Does it require a full-time person now? If not, when does it warrant a full time person?
2. How does this role interface with other parts of 2i2c outside of engineering?
3. Does this role *need* engineering skills, or 

### Time box

I propose that we have a **checkpoint for this role in August 2022** to see how things are going. We can then decide if we want to continue with this pilot, write up what we've learned, and decide on next steps.

### Actions

- [x] Discuss this in the team meeting to see what people think

**Comment by choldgraf** [comment]
### Update: agreement in team meeting

We discussed this issue in today's team meeting, and everybody was +1 on the idea, so let's give it a shot starting today!

There were a few questions raised in the meeting, noted below:

- We should be mindful of whether this role is sustainable at a 50% capacity. @damianavila also helps with our operations and development efforts, and we should make sure that he doesn't feel over-burdened by these extra Project Management duties.
- We should consider any metrics that can be used to measure and track progress for this role. If its goal is to make us more efficient and coordinated, how can we quantify this?
- What's the overlap with our community lead?
  - Product/Community Lead is more external-facing, and the Project Manager is more internal/team-facing.
  - Project Manager should oversee the team "process" and make sure we're following it.

**Comment by damianavila** [comment]
Do we want to keep this one open during the pilot so we can collect our experiences on the road? My first reaction would be YES since we already have a lot of context here. Let me know if you think otherwise.

**Comment by choldgraf** [comment]
+1 to using this issue to collect our experience and what we learn. I think the closing action for this issue would be "summarize and document our learnings, and decide next steps".

**Comment by choldgraf** [comment]
Found a nice blog post about engineering manager pathways and things to look out for: http://charity.wtf/?utm_source=pocket_mylist

Reminds me of some recent conversations I've had with @yuvipanda and @damianavila

**Comment by choldgraf** [comment]
Another job posting about project managers w/ language that we might be able to use as inspiration: https://renci.org/employment/research-project-manager/

**Comment by damianavila** [comment]
Update: These are the steps/changes we have implemented in the last few weeks:

* We have a docs draft PR removing the sprint concept (and replacing it with "cycles") from the documentation: https://github.com/2i2c-org/team-compass/pull/408
* Several backlog and sprint board updates
  * New  columns due/check, iterations, "waiting" status, priority, type
* Maintenance (boards up to date)
  * Keeping track of things we are doing
  * First step to a better prioritization process
* Biweekly mid-cycle written checkpoint
* DIscussions about the support workflow and how the PM should be involved in that process
* Leads involvement
  * 1:1 with Jim on a biweekly basis (off-cycle)
  * Interchange of information about upcoming things in the pipeline
* More board updates
  * No Type and Leads tab
  * reorder tabs
  * added "waiting" tab
  * added minus filters to clean up views
* Thoughts about having 1:1 with all the team members
  * Getting detailed info
  * Helping with prioritization
* Specific 1:1 with Chris (and eventually James) to also exchange info to help in the PM-related process
* New process to go from lead into new hub deployment
  * lead > new hub > assign to damian > get info > reassign to eng
* Talks about the need to do projects in pairs
  * one leading, the other assisting
  * we should be more proactive to embed this idea in our daily practice
* New Must tab
  * Embedded idea that operational tasks may take precedence on top of ongoing projects.  
* Setup 1:1s with eng team members
  * status
  * wishes
  * prioritization
  * evolution

**Comment by choldgraf** [comment]
### Update: Feedback about project manager role

I had a conversation with @damianavila about this last week, and a few points stood out to me that I wanted to note. I'll put them below, but I also want to suggest that we encode this role in our Team Compass rather than in a Google Doc, so that it is easier to find and iterate on.

### It is not clear how much this overlaps with a "team manager"

- In [our job description draft](https://docs.google.com/document/d/1voDLLolN-2szln_KzdpNBH6aaobE8ajcjWYfv4c0DNk/edit#heading=h.dmvmrvfsap2k) we discussed and decided not to call this role a "team manager" because that is different from project management.
- However, we do not have a formally-defined team manager, and so there is confusion about whether this role serves as an "acting team manager" until somebody formally serves in that role.
- It is also not sustainable for @choldgraf to serve as the Engineering Team Manager, he is neither qualified to serve in this role nor does he have the bandwidth now that 2i2c has many moving parts.
- When people apply for [our engineering jobs](https://github.com/2i2c-org/meta/issues/343), they will ask "who will I report to?", is it this role? If not, who is it? 

### Unclear if this role has "assignment" authority for development tasks

- @damianavila feels empowered to assign operational tasks to people (e.g. "we need a hub deployed") but not development priorities and tasks (e.g. we need you to work on improving the terraform infrastructure to do XXX"). This makes it hard for us to be intentional about our balance of operations and development across the team, and also makes it hard to _ensure_ that certain kinds of development tasks get done.

### Hard to coordinate development tasks without a clear definition of our priorities

- It is also hard to assign development tasks without clear development priorities (e.g. this quarter we all agree that XYZ are the most important things to work on).
- We need to define a process or a source of truth for "what are our development priorities" in a way that the project manager can quickly assign tasks to people as-needed.
- In the absence of these priorities, is the Project Manager role empowered to set them?

**Comment by damianavila** [comment]
> However, we do not have a formally-defined team manager, and so there is confusion about whether this role serves as an "acting team manager" until somebody formally serves in that role.

I think there is confusion from the organigram point of view whereas I believe most of the 2i2c actors recognize that I am currently playing (probably in an inefficient way) some of the pieces of this eng team manager role that we have yet undefined.

> @damianavila feels empowered to assign operational tasks to people (e.g. "we need a hub deployed") but not development priorities and tasks (e.g. we need you to work on improving the terraform infrastructure to do XXX"). This makes it hard for us to be intentional about our balance of operations and development across the team, and also makes it hard to ensure that certain kinds of development tasks get done.

We are switching to a new cycle/sprint iteration and an important piece of that transition is the recognition from all the engs that I should feel empowered to assign engs besides the operational tasks. With this agreement in place, which I have already confirmed with the eng team, I think this one will not be an issue in the future.

> Hard to coordinate development tasks without a clear definition of our priorities

We are making progress in this direction as well... with already agreed on 3 main goals for this quarter (Q4 2022): Eng (internal) Documentation improvements, Billing, and Monitoring (alerting). And those goals will be represented in our Eng board so we can use them to pick tasks from those big sets, accordingly to priority, in our cycle/sprint planning meeting.

**Comment by choldgraf** [comment]
### Update: I've refined job descriptions for engineering team lead and for project manager

After a few rounds of feedback from others, I decided to take our original "Project Manager" description and split it into two descriptions, along with some refinement based on our experience so far:

- [Engineering Team Lead](https://docs.google.com/document/d/1m9XFiTNxMwa6gxbCzcobiBLMyQL03VAGUPryyfuKPME/edit?usp=sharing) - this is the top role in an "Engineering Manager" track.
- [Project Manager](https://docs.google.com/document/d/1voDLLolN-2szln_KzdpNBH6aaobE8ajcjWYfv4c0DNk/edit?usp=sharing)

### Next steps?

I suggest that we close this issue after the following actions are taken:

- [ ] @damianavila updates the sprint / team process documents according to the latest developments he described above
- [ ] We iterate on these documents and agree on a good-enough language for now.
- [ ] We document that @damianavila is serving in a hybrid role between these two positions, and also document that we need to decide what path forward we want.

What do others think?

**Comment by damianavila** [comment]
> After a few rounds of feedback from others, I decided to take our original "Project Manager" description and split it into two descriptions, along with some refinement based on our experience so far

Super! Thanks for pushing on that further specification.
In general, I agree with the next steps, but I welcome others' feedback before pushing forward on them.

**Comment by choldgraf** [comment]
The major parts of this role are now captured in our Delivery Manager / Chief of Staff roles, so I'm cross-linking them here and closing out this issue!

https://compass.2i2c.org/engineering/roles/delivery-manager/
